code/datums/elements/item_scaling.dm

```diff
@@ -42,10 +42,10 @@
 	// Make sure overlays also inherit the scaling.
 	ADD_KEEP_TOGETHER(target, ITEM_SCALING_TRAIT)
 
-	// Object scaled when dropped/thrown OR when exiting a storage object.
-	RegisterSignals(target, list(COMSIG_ITEM_DROPPED, COMSIG_ATOM_EXITED), PROC_REF(scale_overworld))
+	// Object scaled when dropped/thrown.
+	RegisterSignal(target, COMSIG_ITEM_DROPPED, PROC_REF(scale_overworld))
 	// Object scaled when placed in an inventory slot OR when entering a storage component.
-	RegisterSignals(target, list(COMSIG_ITEM_EQUIPPED, COMSIG_ATOM_ENTERED), PROC_REF(scale_storage))
+	RegisterSignals(target, list(COMSIG_ITEM_EQUIPPED, COMSIG_ATOM_ENTERING), PROC_REF(scale_storage))
```

Comment:
rather then doing this, you should have dropped and equipped point to their respective procs.
then have entering and exiting point to a proc that checks to see if we're in a turf or not, and changes our scaling accordingly

Reply:
Many thanks for the help on discord, should be much better using `COMSIG_MOVABLE_MOVED` instead.